plotly_express/_core.py

```diff
@@ -869,6 +913,42 @@ def make_figure(args, constructor, trace_patch={}, layout_patch={}):
                         trace.update(marker=dict(color=m.val_map[val]))
                     else:
                         raise
+
+                # Find row for trace, handling facet_row and marginal_x
+                if m.facet == 'row':
+                    if has_marginal_x:
```

Comment:
Is this case for marginals + facets? that's not supported right now in the grouping logic.

Reply:
Yes. With these changes I *think* marginals + facets are working. See the screenshot in the description above.

Reply:
aha I see, that's interesting. Cool that it works! 

I think that for combining them, I was envisioning per-row and per-column marginals rather than per-subplot marginals, which isn't quite as straightforward (although not impossible, just requires a second group-by pass). Thoughts?

Reply:
I liked per subplot marginals, but I suppose per row/col could useful to. Would it make sense for this to be the default and then later add an option to change the behavior?